app/models/public_body.rb

```diff
@@ -202,6 +202,18 @@ def set_api_key!
     self.api_key = SecureRandom.base64(33)
   end
 
+  def self.find_by_name(name)
+    PublicBody.joins(:translations).
+      where("public_body_translations.name = ?", name).
+       first
+  end
+
+  def self.find_by_url_name(url_name)
```

Comment:
Comments for 7813a5d3abd7ae6383c3376926b8f15663759a3c

Explain why we need the `with_translations`. Add specs.